src/Analyser/MutatingScope.php

```diff
@@ -2197,7 +2183,7 @@ public function doNotTreatPhpDocTypesAsCertain(): Scope
 
 	private function promoteNativeTypes(): self
 	{
-		$variableTypes = $this->variableTypes;
+		$expressionTypes = $this->expressionTypes;
 		foreach ($this->nativeExpressionTypes as $expressionType => $type) {
 			if (substr($expressionType, 0, 1) !== '$') {
 				throw new ShouldNotHappenException();
```

Comment:
```suggestion
				continue;
```
that should fix the internal errors most likely :) I'm too much out of context, but it looks like previously this dealt only with variable types and ensured it actually only had variables, while now it can have any expressions inside.